content/blog/vlan-versus-vxlan.md

```diff
@@ -0,0 +1,93 @@
+
+### What is VLAN?
+
+VLAN – Virtual Local Area Network – Layer 2 which segments a physical network into multiple isolated networks. This is achieved by VLAN-ID which can range from <1-4096> which is 12 bit. Different ports on a networking device can be associated to different VLAN-IDs and the networking device would ensure that the incoming/outgoing traffic is forwarded based on the VLAN-ID which is present in the ethernet frames.
+
+So, if the physical network is layer-1, the physical network segmented using VLANs is called layer-2 data link layer.
+
+### What is VXLAN?
+
+VXLAN – Virtual Extensible Local Area Networks – Layer 2 over Layer 3 network
+VXLAN segment is identified by VNI (Virtual Network Identifier) which is 24 bits. The layer-2 ethernet frames are encapsulated by VNI outer headers. Due to this encapsulation, VXLAN is also called tunneling. The VXLAN tunnel end point is called VTEP. VTEPs encapsulate and decapsulates the ethernet frames into VXLAN packets. The intermediate networking devices receiving these VXLAN packets are routed as any other Layer3 Packet. 
+
+### Why do we need VXLAN?
+
+With VLANs being used for segmentation across and within the data centers, spanning tree protocol (STP) will be used for forming loop-free topology. STP disables few links which also results in large number of disabled links.
+This issue would be solved by using a Layer-3 virtual tunnel over the layer-2 network.
+
+A 12-bit VLAN ID is used in the Ethernet data frames to divide the larger Layer 2 network into multiple broadcast domains.  VLANs could be used only for data centers that require fewer than 4094 VLANs. So, scalability issue is resolved using VXLAN.
+
+There could be 24 or 48 servers connected to the Top-of-the-Rack (ToR) switch. Instead of just one MAC address per server link, the switch must learn all the MAC addresses of the virtual machines (VMs) communicating across the servers. This inadequate MAC address table issue would be addressed with the layer3 tunnel over the layer2 frames.
+
+VXLAN is a tunneling mechanism over layer2 ethernet frames. The encapsulation will be known to VTEPs alone and the VMs would not see it.
+
+### VLAN Header in Ethernet Frame
+
+![VLAN Header](/img/picture1.png "VLAN Header")
+
+* The ethernet frame has 4 bytes VLAN tag field to identify the VLAN-ID that the packet originated from. The format of VLAN-tagged frames is defined in IEEE 802.1Q standard.
+* The VLAN TAG has below fields:
+
+  * TPID – By Default VLAN Tagged packets has 0x8100.
+  * Priority – To indicate the 802.1p priority of the frame
+  * CFI – By default it is 0. 0 indicates MAC addresses are in standard format. 1 indicated non-standard format.
+  * VLAN-ID - 12 Bit value to identify the VLAN.
+
+### VXLAN Header in Ethernet Frame
+
+![VXLAN Header](/img/picture2.png "VXLAN Header")
+
+* The entire ethernet frame including DMAC, SMAC and IP-header or IPv6 header and UDP header is encapsulated with four headers:
```

Comment:
Please be consistent with the terms "Ethernet", "header", etc. I grew up with "Ethernet" always being capitalized, as it is the proper name/noun of the technology. "Header", unless it is a proper noun, should just have a lower-case initial letter, as in "header".